app/models/server.rb

```diff
@@ -228,6 +229,29 @@ def collect_supported_tests
     self.save!
   end
 
+  def check_badges
+    badges = []
+    Badge.all.each do |badge|
+      earned = true
+      badge.suites.each do |suite|
```

Comment:
Is it possible to break out of the each blocks at 236 and 241 if earned is set to false?